packages/create-expo-app/src/telemetry.ts

```
@@ -0,0 +1,202 @@
+  if (!analyticsIdentity) {
+    // do not send messages without identities to our backend
+    return;
+  }
+
+  message = { ...message, ...analyticsIdentity };
+  message.type = type;
+
+  if (message.type === 'identify') {
+    message.traits ??= {};
+    message.context ??= {};
+    message.context.traits = message.traits;
+  }
+
+  message.context = {
+    library: {
+      name: library,
+      version,
+    },
+    ...message.context,
+  };
+
+  message._metadata = {
+    nodeVersion: process.versions.node,
+    ...message._metadata,
+  };
+
+  if (!message.originalTimestamp) {
+    message.originalTimestamp = new Date();
+  }
+
+  if (!message.messageId) {
+    // We md5 the messaage to add more randomness. This is primarily meant
+    // for use in the browser where the uuid package falls back to Math.random()
+    // which is not a great source of randomness.
+    // Borrowed from analytics.js (https://github.com/segment-integrations/analytics.js-integration-segmentio/blob/a20d2a2d222aeb3ab2a8c7e72280f1df2618440e/lib/index.js#L255-L256).
+    message.messageId = `node-${crypto
+      .createHash('md5')
+      .update(JSON.stringify(message))
+      .digest('hex')}-${uuidv4()}`;
+  }
+  messageBatch.push(message);
+}
+
+// very barebones implemention...
+// does not support multiple concurrent flushes or large numbers of messages
+export async function flushAsync() {
+  if (!messageBatch.length) {
+    return;
+  }
+
+  const request = {
+    method: 'POST',
+    headers: {
+      accept: 'application/json, text/plain, */*',
+      'content-type': 'application/json;charset=utf-8',
+      'user-agent': `${library}/${version}`,
+      authorization: 'Basic ' + Buffer.from(`${xdlUnifiedWriteKey}:`).toString('base64'),
+    },
+    body: JSON.stringify({
+      batch: messageBatch.map(message => ({ ...message, sentAt: new Date() })),
+      sentAt: new Date(),
+    }),
+  };
+  await fetch(analyticsEndpoint, request);
```

do we want to handle errors / failures? for example, if the user is offline or our endpoint is down. we could wrap this in a try catch and suppress errors, etc